src/ntlm/mod.rs

```diff
@@ -228,11 +229,66 @@ impl SspiImpl for Ntlm {
         })
     }
 
+
+
+    #[instrument(level = "debug", ret, fields(state = ?self.state), skip(self, builder))]
+    fn accept_security_context_impl(
```

Comment:
separate impl from trait method allows poping up Error and wrap in inside Generator as
type
```
 Generator <'a,_,_,Result<T,E>>
```

nested result of type 
```
Result< Generator <'a,_,_,Result<T,E>>,E>
```
is ugly